ykbh/src/lib.rs

```diff
@@ -187,7 +188,7 @@ impl SIRInterpreter {
         }
     }
 
-    pub unsafe fn interpret(&mut self, body: ykpack::Body) {
+    pub unsafe fn interpret(&mut self, body: Arc<ykpack::Body>) {
```

Comment:
It *might* be better to have this be `&ykpack::Body`? That might not work for other reasons though.

Reply:
Does `&*` at the call site work lukas?

Reply:
Unfortunately not, since I also need to hold on to those references for a while in a vector: https://github.com/ptersilie/yk/blob/sinterp_calls/ykbh/src/lib.rs#L234

Reply:
OK, maybe not for this PR, but for soon: it feels a bit weird that `interpret` is storing bodies (or, at least, now `Arc<Body>`, which might not actually be so bad). We might want to have a deeper think about that.

Reply:
> Does `&*` at the call site work lukas?

This needs to be `body: ykpack::Body`. It can't be a reference. Not sure if I could just move the `body` out like it did before.

Reply:
Gah.

That `Arc` is annoying me too. I have a branch which uses a `SyncOnceCell` instead of an `Arc`, but I'm still evaluating it...

Reply:
> OK, maybe not for this PR, but for soon: it feels a bit weird that `interpret` is storing bodies (or, at least, now `Arc<Body>`, which might not actually be so bad). We might want to have a deeper think about that.

I need to store the bodies to avoid having a recursion in `interpret` when interpreting nested calls. See https://github.com/softdevteam/yk/pull/176/commits/016af9814821733a11030eb15358c17392ea9720

Reply:
Alternatively, I could store the function name, and then pull body back out of SIR when returning to an older frame.

Reply:
I'm not disputing the usecase, just the mechanism by which we code it up. Anyway, it's not for this PR.